src/components/connection_handler/include/connection_handler/heartbeat_monitor.h

```diff
@@ -111,7 +119,7 @@ class HeartBeatMonitor : public threads::ThreadDelegate {
   typedef std::map<uint8_t, SessionState> SessionMap;
   SessionMap sessions_;
 
-  sync_primitives::Lock sessions_list_lock_;  // recurcive
+  mutable sync_primitives::Lock sessions_list_lock_;  // recurcive
```

Comment:
@AKalinich-Luxoft `recursive`

Reply:
1. Fixed in https://github.com/AKalinich-Luxoft/sdl_core/pull/2/commits/d3bca99da653e13fdb1e7986d4ea6f09ca37ba11
2. Because IsSessionHeartbeatTracked is a const function, however we need a mutex for a safe access to a sessions map even in a const method so it becomes a mutable.